counter/src/main/java/org/infinispan/counter/impl/listener/CounterManagerNotificationManager.java

```diff
@@ -119,7 +120,8 @@ public synchronized void listenOn(Cache<CounterKey, CounterValue> cache) throws
          topologyListener.register(cache);
       }
       if (!listenersRegistered) {
-         this.cache.addListener(valueListener, CounterKeyFilter.getInstance());
+         this.cache.addFilteredListener(valueListener, null, null,
```

Comment:
This looks wrong, we aren't using the filter anymore.

Reply:
Also the override to specify the events is more for remote, not for embedded as we can properly ascertain this from the annotations of the methods.

Reply:
> This looks wrong, we aren't using the filter anymore.

Previously we were using `CounterKeyFilter` whose accept method is `return key instanceof CounterKey;` which will always be the case for the counter cache. As `CounterManagerNotificationManager` is in a `.impl` package this isn't API, so AFAIK it will work for our internal needs.